### ISSUE TITLE
[Append Clipboard] ...

### ISSUE THREAD

**Issue by fbracht** [raycast/extensions#4352]
### Extension

https://www.raycast.com/koinzhang/append-clipboard

#### Error: 
```
Error: Cannot copy selected text from frontmost application.
```

Sorry, I don't know how to describe the issue in technical terms, but in summary and in layman's terms here's what happens:

Roughly every time I start using my macbook after it has been in sleep, it happens that I'll try to use Raycast and my use will get blocked by a Raycast error/bug screen. I hit my Raycast global shortcut and start typing, but I hear macOS system sounds that tell me my keystrokes are not going through. Then I'll look up to the screen and I'll see this error/bug screen on Raycast instead of the expected screen (the root search). The error screen in Raycast has the Append to Clipboard extension label on the bottom left, and hitting Cmd+K I was able to select Report Bug and that's how I got to this page. 

This error screen goes away if I hit Esc to return Raycast to the root search screen, so it's not a problem. (It's just an annoyance, because every time this happens I'll have to hit Esc and then type my query again.)

And, again, this seems to happen semi-consistently the first time I try to use Raycast after my macbook has been asleep for any significant period of time.

I'll try to update this bug report with a screenshot next time it happens.

Thank you.

**Comment by fbracht** [comment]
![CleanShot 2023-01-20 at 15 45 07@2x](https://user-images.githubusercontent.com/8870859/213780839-60bc8076-6ef8-4c66-a4af-695b523fb799.png)

**Comment by fbracht** [comment]
I actually figured out what exactly is causing this! It's pretty simple. 

I have a global shortcut to the Append Text to Clipboard action. It's a double-tap of the Cmd key. If I do this while there's nothing selected that could be appended to clipboard, then the next Raycast activation will bring this screen up and prevent me from typing anything before hitting Esc to dismiss. 

The solution doesn't seem much more complicated than making sure the extension is able to gracefully handle not being passed any selection data by the command — by either simply ignoring the command or surfacing a "nothing selected to append" message via toast.

**Comment by fbracht** [comment]
(I have no idea why I though there was any correlation between my mac having woken up from sleep and this issue happening. There doesn't seem to be any.)

**Comment by koinzhang** [comment]
Hi @fbracht Thank you for your feedback, I have noted the issue and submitted a PR #4536 for update

**Comment by koinzhang** [comment]
@raycastbot close this issue
